From 02100134eb063fbba4f1fc28382fae6d657b0e51 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Wed, 22 Jun 2022 10:43:26 -0700
Subject: [PATCH 01/15] ntb_perf: Fix 64-bit division on 32-bit architectures

When compiling for a 32-bit architecture, such as arm, an error occurs
during modpost:

  ERROR: modpost: "__aeabi_uldivmod" [drivers/ntb/test/ntb_perf.ko] undefined!

The tries member of struct perf_thread is u64 so a 64-bit division
helper is needed. Use div_u64_rem() to get the remainder of the division
so that it can be checked against zero.

Fixes: dc150dfb081f ("ntb_perf: extend with burst latency measurement")
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20220622174326.1832697-1-nathan@kernel.org
---
 drivers/ntb/test/ntb_perf.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/ntb/test/ntb_perf.c b/drivers/ntb/test/ntb_perf.c
index 536fab0030f3..af7d3de65eb3 100644
--- a/drivers/ntb/test/ntb_perf.c
+++ b/drivers/ntb/test/ntb_perf.c
@@ -1118,6 +1118,7 @@ static int perf_run_latency(struct perf_thread *pthr)
 	void __iomem *flt_dst, *bnd_dst;
 	void *flt_src;
 	u64 stop_at;
+	u32 rem;
 	int ret;
 
 	pthr->tries = 0;
@@ -1146,7 +1147,8 @@ static int perf_run_latency(struct perf_thread *pthr)
 		}
 
 		/* Avoid processor soft lock-ups */
-		if (!(pthr->tries % RESCHEDULE_RATIO))
+		div_u64_rem(pthr->tries, RESCHEDULE_RATIO, &rem);
+		if (!rem)
 			schedule();
 	}
 

base-commit: 4662b7adea50bb62e993a67f611f3be625d3df0d
-- 
2.37.1

